agent/structs/connect_proxy_config.go

```diff
@@ -333,6 +333,9 @@ func (u *Upstream) Validate() error {
 	if u.DestinationName == "" {
 		return fmt.Errorf("upstream destination name cannot be empty")
 	}
+	if u.DestinationName == WildcardSpecifier && !u.CentrallyConfigured {
```

Comment:
Why gate this on CentrallyConfigured? A service name of `*` isn't going to result in SNI stuff in xDS that functions so it should be safe to prohibit all the time.